app/discord_client.py

```diff
@@ -0,0 +1,7 @@
+class DiscordClient():
+    def __init__(self, client, bot_secret_key):
+        self._client = client
+        self._bot_secret_key = bot_secret_key
+
+    def run(self):
+        self._client.run(self._bot_secret_key)
```

Comment:
```suggestion
from discord import Client

class DiscordClient(Client):
    pass
```
Ça ne serait pas plus simple d’étendre directement discord.Client ?

Reply:
Non, parce que si tu fais ça, tu deviens complètement dépendant de l'implémentation de discord.Client alors que le but ici est de rendre DiscordClient plus facilement testable.
Si discord.py est mis à jour plus tard et casse certains comportements, tu le retrouveras dans tout ton code qui utilise les méthodes directement et il te faudra  modifier ça un peu partout. Alors qu'ici, il te suffit de changer l'implémentation de DiscordClient.

Evidemment, l'intérêt ici est encore peu visible  puisque le bot ne fait rien de particulier encore.

Reply:
j'approuve le choix de la composition, c'est plus prudent (et en plus, ça a le mérite d'être plus explicite)